movieDetails.js

```diff
@@ -0,0 +1,242 @@
+
+  const moviePoster = document.createElement("img");
+  moviePoster.src = movie.poster_url;
+  moviePoster.alt = `Poster for ${movie.title}`;
+  moviePoster.classList.add("movie-poster");
+
+  const movieDetailsSide = document.createElement("div");
+  movieDetailsSide.classList.add("movie-details-side");
+
+  const movieYear = document.createElement("p");
+  movieYear.classList.add("movie-details");
+  movieYear.innerHTML = `<strong>Year:</strong> ${movie.movie_year}`;
+
+  const movieDescription = document.createElement("p");
+  movieDescription.classList.add("movie-description");
+  movieDescription.textContent = movie.description;
+
+  movieDetailsSide.append(movieYear, movieDescription);
+  movieCardLayout.append(moviePoster, movieDetailsSide);
+
+  const movieDirector = document.createElement("p");
+  movieDirector.classList.add("movie-director");
+  movieDirector.textContent = `Director: ${movie.director}`;
+
+  const movieExtraInfo = document.createElement("div");
+  movieExtraInfo.classList.add("movie-extra-info");
+  movieExtraInfo.dataset.movieId = movie.id;
+
+  const moviePrice = document.createElement("p");
+  moviePrice.classList.add("movie-price");
+  moviePrice.textContent = `DKK${movie.price}`;
+
+  const sendMessage = document.createElement("div");
+  sendMessage.classList.add("movie-info-icon");
+
+  const movieRatings = document.createElement("div");
+  movieRatings.classList.add("movie-ratings");
+
+  const messageIcon = document.createElement("i");
+  messageIcon.classList.add("fa-regular", "fa-message");
+  sendMessage.appendChild(messageIcon);
+
+  for (let i = 0; i < 5; i++) {
+    const ratingButton = document.createElement("button");
+    ratingButton.classList.add("rating");
+
+    const starIcon = document.createElement("i");
+    starIcon.classList.add("fa-regular", "fa-star");
+
+    ratingButton.appendChild(starIcon);
+    movieRatings.appendChild(ratingButton);
+
+    ratingButton.addEventListener("click", () => {
+      movie.rating = i + 1;
+      updateStars(movieRatings, movie.rating);
+    });
+  }
+
+  movieExtraInfo.append(moviePrice, movieRatings, sendMessage);
+  movieCard.append(movieTitle, movieCardLayout, movieDirector, movieExtraInfo);
+
+  sendMessage.addEventListener("click", function (e) {
+    e.stopPropagation();
+    openCommentsPopup(movie);
+  });
+
+  movieCardLayout.addEventListener("click", function () {
+    openMovieDetailsPopup(movie);
+  });
+
+  updateStars(movieRatings, movie.rating || 0);
+
+  return movieCard;
+}
+
+// Update star rating
+function updateStars(ratingContainer, rating) {
+  const stars = ratingContainer.querySelectorAll(".rating i");
+  stars.forEach((star, index) => {
+    if (index < rating) {
+      star.classList.add("fa-solid");
+      star.classList.remove("fa-regular");
+    } else {
+      star.classList.add("fa-regular");
+      star.classList.remove("fa-solid");
+    }
+  });
+}
+
+// Popup and comment functionality 
+function openCommentsPopup(movie) {
```

Comment:
Same comment as my previous one. You can split this function into smaller functions to make your code more readable.